resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/internal/properties/Bug468FunctionalTest.java

```diff
@@ -0,0 +1,125 @@
+ *
+ * Contributors:
+ *     ETAS GmbH - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.core.tests.internal.properties;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import org.eclipse.core.internal.properties.PropertyManager2;
+import org.eclipse.core.internal.resources.Workspace;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.ICoreRunnable;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.QualifiedName;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+//Test case for GitHub Issue 468
+@RunWith(JUnit4.class)
+public class Bug468FunctionalTest {
+
+	private static final String TO_BE_DELETED_FILE_NAME_PREFIX = "file_";
+
+	private static final String TEMP_FOLDER_NAME = "temp";
+
+	private IProject project;
+
+	/*
+	 * @throws Exception If anything goes wrong during the set up.
+	 */
+
+	@Before
+	public void setUp() throws Exception {
+		ResourcesPlugin.getWorkspace().run((ICoreRunnable) monitor -> {
+
+			createTestProject();
+			IFolder tempFolder = Bug468FunctionalTest.this.project.getFolder(TEMP_FOLDER_NAME);
+			tempFolder.create(true, true, new NullProgressMonitor());
+			createFile(tempFolder, TO_BE_DELETED_FILE_NAME_PREFIX + 0);
+
+		}, this.project, IWorkspace.AVOID_UPDATE, new NullProgressMonitor());
+	}
+
+	private void createTestProject() throws CoreException {
+		Bug468FunctionalTest.this.project = ResourcesPlugin.getWorkspace().getRoot()
+				.getProject("TestProject");
+		Bug468FunctionalTest.this.project.create(new NullProgressMonitor());
+		Bug468FunctionalTest.this.project.open(new NullProgressMonitor());
+	}
+	private IFile createFile(final IFolder parent, final String fileName) throws CoreException {
+		IFile file = parent.getFile(fileName);
+		InputStream source = new ByteArrayInputStream(file.getName().getBytes());
+		file.create(source, true, new NullProgressMonitor());
+		file.setPersistentProperty(new QualifiedName(this.getClass().getName(), file.getName()), file.getName());
+		return file;
+	}
+
+	/**
+	 * Deletes the project
+	 *
+	 * @throws Exception
+	 *             if any exception happens during the deleting of the project.
+	 */
+	@After
+	public void tearDown() throws Exception {
+		this.project.delete(true, new NullProgressMonitor());
+	}
+
+	Workspace ws;
+	PropertyManager2 manager;
```

Comment:
Please remove these fields and just declare the variables inside the `test` method.